packages/core/src/supertokens.ts

```diff
@@ -140,15 +140,17 @@ publicDataStore.initialize()
 
 export const useSession = () => {
   const [publicData, setPublicData] = useState(emptyPublicData)
+  const [isReady, setIsReady] = useState(false)
 
   useEffect(() => {
```

Comment:
Also as a small optimization, let's add this

```ts
const IS_SERVER = typeof window === 'undefined'

// React currently throws a warning when using useLayoutEffect on the server.
// To get around it, we can conditionally useEffect on the server (no-op) and
// useLayoutEffect in the browser.
const useIsomorphicLayoutEffect = IS_SERVER ? useEffect : useLayoutEffect
```

and then this:

```suggestion
  useIsomorphicLayoutEffect(() => {
```

Reply:
hey @flybayer , sounds good. How about we define `useIsomorphicLayoutEffect` inside `/utils/hooks.ts` and export it from there, in case it needs to be used somewhere else? Or would you rather just colocate it next to the `useSession` definition?

Reply:
Yeah that's fine to put it in utils/hooks :)